internal/rust/parser/src/strconv.rs

```diff
@@ -0,0 +1,198 @@
+use chrono::prelude::*;
+use chrono::FixedOffset;
+
+use std::iter::Peekable;
+use std::str::{CharIndices, Chars};
+
+use regex::Regex;
+
+pub fn parse_string(lit: &str) -> Result<String, String> {
+    if lit.len() < 2 || lit.chars().next().unwrap() != '"' || lit.chars().last().unwrap() != '"' {
+        return Err("invalid string literal".to_string());
+    }
+    let mut s = Vec::with_capacity(lit.len());
+    let mut chars = lit.char_indices();
+    let last = lit.len() - 1;
+    loop {
+        match chars.next() {
+            Some((i, c)) => {
+                if i != 0 && i != last {
+                    match c {
+                        '\\' => match push_unescaped(&mut s, &mut chars) {
+                            Err(e) => return Err(e.to_string()),
+                            _ => (),
+                        },
+                        // this char can have any byte length
+                        _ => s.extend_from_slice(c.to_string().as_bytes()),
+                    }
+                }
+            }
+            None => break,
+        }
+    }
+    let converted = std::str::from_utf8(&s);
+    match converted {
+        Ok(s) => Ok(s.to_string()),
+        Err(e) => Err(e.to_string()),
+    }
+}
+
+fn push_unescaped(s: &mut Vec<u8>, chars: &mut CharIndices) -> Result<(), String> {
+    match chars.next() {
+        Some((_, c)) => match c {
+            'n' => s.push(b'\n'),
+            'r' => s.push(b'\r'),
+            't' => s.push(b'\t'),
+            '\\' => s.push(b'\\'),
+            '"' => s.push(b'"'),
+            'x' => {
+                let ch1 = match chars.next() {
+                    Some((_, c)) => c,
+                    None => return Err(r#"\x followed by 0 char, must be 2"#.to_string()),
+                };
+                let ch2 = match chars.next() {
+                    Some((_, c)) => c,
+                    None => return Err(r#"\x followed by 1 char, must be 2"#.to_string()),
+                };
+                let b1 = to_byte(ch1);
+                let b2 = to_byte(ch2);
+                if b1.is_none() || b2.is_none() {
+                    return Err("invalid byte value".to_string());
+                }
+                let b = (b1.unwrap() << 4) | b2.unwrap();
+                s.push(b);
+            }
+            _ => return Err(format!("invalid escape character {}", c)),
+        },
+        None => return Err("invalid escape sequence".to_string()),
+    };
+    Ok(())
+}
+
+fn to_byte(c: char) -> Option<u8> {
+    match c {
+        '0'...'9' => Some(c as u8 - b'0'),
+        'a'...'f' => Some(c as u8 - b'a' + 10),
+        'A'...'F' => Some(c as u8 - b'A' + 10),
```

Comment:
My compiler gave me a warning about this syntax being deprecated and it suggests something like `..=`

Reply:
here is the complete compile warning:

```
warning: `...` range patterns are deprecated
  --> src/strconv.rs:76:12
   |
76 |         'A'...'F' => Some(c as u8 - b'A' + 10),
   |            ^^^ help: use `..=` for an inclusive range

```

Reply:
I don't get those warning event after `rustup update`.  
Do you get those in `cargo build`?

Reply:
Anyways I updated and everything works... Maybe I'm running the wrong command when building.

Reply:
Maybe do `cargo clean` and then `cargo build`? I noticed that the warnings only show up when it actually re compiles those bits, which isn't often.

Reply:
Yeah, I tried `clean` + `build` but I only got warnings on the unused C variables 😓